common/src/main/java/com/wynntils/screens/partymanagement/widgets/PartyPlayer.java

```diff
@@ -0,0 +1,123 @@
+import com.wynntils.core.components.Models;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+
+public class PartyPlayer extends AbstractWidget {
+
+    private final String playerName;
+    private final boolean isOffline;
+    private final Button promoteButton;
+    private final Button kickButton;
+    private final Button disbandButton;
+
+    public PartyPlayer(int x, int y, int width, int height, String playerName, boolean isOffline) {
+        super(x, y, width, height, Component.literal(playerName));
+        this.playerName = playerName;
+        this.isOffline = isOffline;
+        this.promoteButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.partyManagementGui.promote"),
+                        (button) -> promoteToLeader(playerName))
+                .pos(this.getX() + 240, this.getY())
+                .size(50, 20)
+                .build();
+        this.kickButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.partyManagementGui.kick"),
+                        (button) -> kickFromParty(playerName))
+                .pos(this.getX() + 292, this.getY())
+                .size(50, 20)
+                .build();
+        this.disbandButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.partyManagementGui.disband"),
+                        (button) -> disbandParty())
+                .pos(this.getX() + 292, this.getY())
+                .size(50, 20)
+                .build();
+    }
+
+    @Override
+    public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        ResourceLocation skin =
+                McUtils.mc().getConnection().getPlayerInfo(playerName).getSkinLocation();
+        // head rendering
+        RenderUtils.drawTexturedRect(poseStack, skin, this.getX(), this.getY(), 8, 16, 16, 8, 8, 8, 8, 64, 64);
+        // hat rendering
+        RenderUtils.drawTexturedRect(poseStack, skin, this.getX(), this.getY(), 1, 16, 16, 40, 8, 8, 8, 64, 64);
+
+        // name rendering
+        CustomColor color = CommonColors.WHITE;
+        if (playerName.equals(Models.Party.getPartyLeader())) {
+            color = CommonColors.YELLOW;
+        } else if (Models.Friends.getFriends().contains(playerName)) {
+            color = CommonColors.GREEN;
+        }
```

Comment:
No color for guild? Also, shouldn't friendship take priority in coloring? (Sorry I am not sure where this is used exactly, just a "blind" suggestion)

Reply:
Guild is not actually suggested in legacy, nor is it highlighted in the member list. I am not sure how often people party with their guild as I am not really in a large guild, but we can look into this if there is demand.

The leader/friend coloring is used in the member list, where leader is yellow and friends are green. I think it is more important to always have a leader visible, so friends would be lower priority.